include/fi_atom.h

```diff
@@ -53,150 +53,149 @@ extern "C" {
 
 
 #if ENABLE_DEBUG
+#define ATOMIC_DEF_INIT int is_initialized
 #define ATOMIC_IS_INITIALIZED(atomic) assert(atomic->is_initialized)
+#define ATOMIC_INIT(atomic) atomic->is_initialized = 1
 #else
+#define ATOMIC_DEF_INIT
 #define ATOMIC_IS_INITIALIZED(atomic)
+#define ATOMIC_INIT(atomic)
 #endif
 
 #ifdef HAVE_ATOMICS
-typedef struct {
-    atomic_int val;
-#if ENABLE_DEBUG
-    int is_initialized;
-#endif
-} atomic_t;
-
-static inline int atomic_inc(atomic_t *atomic)
-{
-	ATOMIC_IS_INITIALIZED(atomic);
-	return atomic_fetch_add_explicit(&atomic->val, 1, memory_order_acq_rel) + 1;
-}
-
-static inline int atomic_dec(atomic_t *atomic)
-{
-	ATOMIC_IS_INITIALIZED(atomic);
-	return atomic_fetch_sub_explicit(&atomic->val, 1, memory_order_acq_rel) - 1;
-}
-
-static inline int atomic_set(atomic_t *atomic, int value)
-{
-	ATOMIC_IS_INITIALIZED(atomic);
-	atomic_store(&atomic->val, value);
-	return value;
-}
-
-static inline int atomic_get(atomic_t *atomic)
-{
-	ATOMIC_IS_INITIALIZED(atomic);
-	return atomic_load(&atomic->val);
-}
-
-/* avoid using "atomic_init" so we don't conflict with symbol/macro from stdatomic.h */
-static inline void atomic_initialize(atomic_t *atomic, int value)
-{
-	atomic_init(&atomic->val, value);
-#if ENABLE_DEBUG
-	atomic->is_initialized = 1;
-#endif
-}
-
-static inline int atomic_add(atomic_t *atomic, int val)
-{
-	ATOMIC_IS_INITIALIZED(atomic);
-	return atomic_fetch_add_explicit(&atomic->val,
-			val, memory_order_acq_rel) + 1;
-}
-
-static inline int atomic_sub(atomic_t *atomic, int val)
-{
-	ATOMIC_IS_INITIALIZED(atomic);
-	return atomic_fetch_sub_explicit(&atomic->val,
-			val, memory_order_acq_rel) - 1;
-}
+	typedef atomic_int	atomic_int32_t;
```

Comment:
`atomic_int32_t` is a type defined by C++ `std::atomic` probably not a good idea to give it the same name. I think this should be `ofi_atomic_int32_t` or something with an `ofi_` prefix indicating it's in the Libfabric namespace.